### PR TITLE
Add new advertising UI layout

### PR DESCRIPTION
**Pull request by stonko1994**
## Description

This PR adds the new UI layout for Ad playback according to the current design. The UI looks the same for small screen and default.

Screenshot 2024-12-18 at 13 57 22

## Checklist (for PR submitter and reviewers)

- [x] `CHANGELOG` entry **A general entry will be added at the end**

### CODE REVIEW

File: src/scss/skin-super-modern/_skin-ads.scss [hunk 1]
```diff
@@ -1,40 +1,26 @@
...
+  @import 'components/adstatusoverlay';
+
+  .#{$prefix}-ui-seekbar {
+    .#{$prefix}-seekbar,
+    .#{$prefix}-seekbar-bars,
+    .#{$prefix}-seekbar-bars > * {
+      pointer-events: none;
     }
 
-    .#{$prefix}-ui-button-ad-skip {
```
Comment (stonko1994) [inline src/scss/skin-super-modern/_skin-ads.scss:20]: The styling for the ad-skip-button was extracted into a separate file.

File: src/ts/components/seekbar.ts [hunk 2]
```diff
@@ -569,6 +569,13 @@ export class SeekBar extends Component<SeekBarConfig> {
         return;
       }
 
+      // Reset the currentTimeSeekBar and set the position to 0 if the player has no duration
+      if (this.player.getDuration() === 0) {
+        this.setPlaybackPosition(0);
+        currentTimeSeekBar = 0;
+        return;
+      }
```
Comment (stonko1994) [inline src/ts/components/seekbar.ts:577]: This was no problem before as there was no seekbar visible during ad playback.

File: src/scss/skin-super-modern/components/_adskipbutton.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  @extend %ui-button;
+
+  background-color: transparentize(#000, 0.75);
+  border-radius: 20px;
+  padding: 0em 1em;
+  min-width: fit-content;
+
+  .#{$prefix}-label {
+    display: inline-block;
+    color: $color-ads;
```
Comment (bitmovin-kenny) [inline src/scss/skin-super-modern/components/_adskipbutton.scss:13]: TBH i find the yellow text hard to read, are we sure we want to stick with that yellow?

**Review by bitmovin-kenny — CHANGES_REQUESTED** [review]
Looks nice 💪
But there is a one pixel black border on the left side of the player during ad playback, on a white background even I noticed it:
image

**Comment by stonko1994** [comment]
> Looks nice 💪 But there is a one pixel black border on the left side of the player during ad playback, on a white background even I noticed it: ...

That bothers me since 6 1/2 years 😄 Is also there on development so nothing new. I'll note that and maybe revisit this if time allows.

Screenshot 2024-12-20 at 09 12 13

File: src/scss/skin-super-modern/components/_adskipbutton.scss (hunk 3 above)
Comment (stonko1994) [reply src/scss/skin-super-modern/components/_adskipbutton.scss:13]: Noted. I'll take that feedback back to the design process.

**Review by bitmovin-kenny — APPROVED** [review]
Damn, it's really in dev too

File: src/scss/skin-super-modern/components/_adskipbutton.scss (hunk 3 above)
Comment (bitmovin-kenny) [reply src/scss/skin-super-modern/components/_adskipbutton.scss:13]: Don't your eyes hurt when looking at the above screenshot and trying to figure out how may seconds there are left until skip?

**Pull request merged**